docs/userstories/egress-control-in-anp/egress-control-cluster-nodes.md

```diff
@@ -0,0 +1,42 @@
+
+### Summary
+
+1. **As a** cluster administrator **I want** to easily block access from
+cluster workloads to k8s-apiservers running on cluster nodes (host-net pods).
```

Comment:
> I think apiservers should be their own user story, because
> 
> * They might be running on non-node hosts, so any solution that involves a podSelector or nodeSelector won't work for all cases.

ack that makes sense, I can split this one as separate use case and keep this user story more generic.
By non-node hosts you mean some random VMs or bootstrap machines or something that are neither recognized as pods nor nodes in the cluster right?

> * There might be external LoadBalancers pointing to the apiserver endpoints as well, and if the admin is blocking access via the clusterIP / endpoints, then they probably want to block access via external LBs too.

+1 yeah forgot about this detail, I guess now that I think about this more, deff warrants a new user story because this falls under the umbrella of "services backed by host-networked pods" OR "services backed by non-node/non-pod entities" of sorts versus we should keep this story as about host-networked pods alone a.k.a nodes that are part of cluster.

> 
> That might be something to implement slightly later than the basic egress stuff, but I think it's something worth tracking.
> 
> (So, IMO, rewrite this user story to be about generic hostNetwork pods and node services, and add another user story about apiservers.)

++ will do.

Reply:
>  By non-node hosts you mean some random VMs or bootstrap machines or something that are neither recognized as pods nor nodes in the cluster right?

"something that are neither recognized as pods nor nodes in the cluster", yes. There are no requirements on where the apiservers run. In (ordinary) OpenShift we run them as static pods inside the cluster, but I think that's pretty uncommon for kubernetes clusters in general (since it requires all the trickiness with bootstrapping).
(Even in HyperShift it doesn't work that way.)

Reply:
done. I didn't add it as a new user story, but I mentioned this use case in the non-goals section - so that we still track it but it won't be a part of this NPEP.